### PR TITLE
Bump versions after for preview 3

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Comment by jianghaolu** [comment]
@conniey  coming

**Comment by jianghaolu** [comment]
@conniey I may not be able to do due to potential warnings/errors but let me give it a try

File: eng/spotbugs-aggregate-report/pom.xml [hunk 1]
```diff
@@ -23,15 +23,13 @@
 
     <!-- Track 2 client library versions -->
     <azure-core.version>1.0.0-preview.3</azure-core.version>
```
Comment (conniey) [inline eng/spotbugs-aggregate-report/pom.xml]: should this be -preview.4 since you've updated the version in the pom.xml

**Comment by alzimmermsft** [comment]
Don't forget to update the other {service-name}Configuration files that determine the User-Agent

File: eng/spotbugs-aggregate-report/pom.xml (hunk 1 above)
Comment (jianghaolu) [reply eng/spotbugs-aggregate-report/pom.xml]: Yes!

File: eng/spotbugs-aggregate-report/pom.xml [hunk 2]
```diff
@@ -22,16 +22,14 @@
...
-    <azure-data-appconfiguration.version>1.0.0-preview.2</azure-data-appconfiguration.version>
-    <azure-identity.version>1.0.0-preview.1</azure-identity.version>
-    <azure-keyvault.version>4.0.0-preview.1</azure-keyvault.version>
-    <azure-identity.version>1.0.0-preview.2</azure-identity.version>
-    <azure-keyvault.version>4.0.0-preview.1</azure-keyvault.version>
-    <azure-messaging-eventhubs.version>5.0.0-preview.2</azure-messaging-eventhubs.version>
-    <azure-storage-blob.version>12.0.0-preview.2</azure-storage-blob.version>
-    <azure-storage-file.version>12.0.0-preview.2</azure-storage-file.version>
-    <azure-storage-queue.version>12.0.0-preview.2</azure-storage-queue.version>
+    <azure-core.version>1.0.0-preview.4</azure-core.version>
```
Comment (srnagar) [inline eng/spotbugs-aggregate-report/pom.xml]: Need to update versions in `eng/jacoco-test-coverage/pom.xml` as well.

**Pull request merged**